### PR TITLE
Subtasks handler in wait_for_completed

### PR DESCRIPTION
**Pull request by danielui**
# Pull Request summary:
- Add subtasks handler to wait_for_completed.
- Changed Existing API's methods using wait_for_completed, because wait_for_completed starts to returns TaskResult(bool, List[SubTaskData]) - bool is True if all subtasks are success, or returns False if at least one is failed
- Change exception blocks into conditional statements - no more looking for first element in list, so IndexError will not be raised
- Use custom exceptions instead of built-ins
- Use pydantic data model intsead attrs lib

# Checklist:
- [X] Make sure to run pre-commit before commiting changes
- [X] Make sure all checks have passed
- [X] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [X] Make sure you test the changes

### CODE REVIEW

File: vmngclient/api/template_api.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def get_device_specific_variables(name: str):
         endpoint = "/dataservice/template/device/config/attachfeature"
         logger.info(f"Attaching a template: {name} to the device: {device.hostname}.")
         response = self.session.post(url=endpoint, json=payload).json()
-        task = wait_for_completed(session=self.session, action_id=response["id"])
+        task = wait_for_completed(session=self.session, action_id=response["id"])[0]
```
Comment (tomaszwarda) [inline vmngclient/api/template_api.py:189]:
```suggestion
        task = wait_for_completed(session=self.session, action_id=response["id"]).single_or_default()
```

File: vmngclient/api/task_status_api.py [hunk 2]
```diff
@@ -88,10 +94,12 @@ def wait_for_completed(
         task (TaskStatus):
     """
```
Comment (tomaszwarda) [inline vmngclient/api/task_status_api.py:95]: Update `Args` in docstring to new arguments in function.

File: vmngclient/api/template_api.py (hunk 1 above)
Comment (JimOverholt) [reply vmngclient/api/template_api.py:189]: OK, at first I was confused with .singe or default, but it also made sense at the same time.
Now that I see it in code it it's obvious...
`.single_or_default() == [0]`

**Review by sbasan — DISMISSED** [review]
lgtm

**Review by JimOverholt — DISMISSED** [review]

File: vmngclient/api/task_status_api.py [hunk 3]
```diff
@@ -4,151 +4,237 @@
...
 
         Args:
-            status (str): status of task
-            status_id (str): status id of task
-            activity (str): activity text
+            session (vManageSession): session
 
         Returns:
-            bool: False if condition is met
+        List[str]: active tasks id's
```
Comment (sbasan) [inline vmngclient/api/task_status_api.py:76]: docstring: update with correct return type = TasksData

File: vmngclient/api/task_status_api.py [hunk 4]
```diff
@@ -4,151 +4,237 @@
...
-        wait=wait_fixed(interval_seconds),
-        stop=stop_after_attempt(int(timeout_seconds / interval_seconds)),
-        retry=retry_if_result(check_status),
-        retry_error_callback=log_exception,
-    )
-    def wait_for_action_finish() -> TaskStatus:
-        """
-        Keep asking for task status, status_id,
-        activity(optional), utill check_status is True
+    def get_task_data(self, delay_seconds: int = 5) -> List[SubTaskData]:
```
Comment (sbasan) [inline vmngclient/api/task_status_api.py:82]: add docstring

**Review by sbasan — CHANGES_REQUESTED** [review]
docstrings are not up-to-date

**Review by JimOverholt — APPROVED** [review]

**Pull request merged**